expression/builtin_time.go

```diff
@@ -2553,8 +2553,8 @@ func (du *baseDateArithmitical) getIntervalFromString(ctx sessionctx.Context, ar
 	if isNull || err != nil {
 		return "", true, errors.Trace(err)
 	}
-	// unit "DAY" has to be specially handled.
-	if strings.ToLower(unit) == "day" {
+	// unit "DAY" and "HOUR" has to be specially handled.
+	if strings.ToLower(unit) == "day" || strings.ToLower(unit) == "hour" {
```

Comment:
We can just call ToLower once.

Reply:
it's mean we need a temp variable for ToLower?

Reply:
@ngaut already remove redundant ToLower call.